src/raps/actions/swap.ts

```diff
@@ -251,9 +318,10 @@ export const swap = async ({
 
   let gasLimit;
   try {
-    gasLimit = await estimateSwapGasLimit({
+    gasLimit = await estimateUnlockAndSwap({
```

Comment:
main change is to use this function instead, which takes into account the metadata estimations